layrry-config/src/main/java/org/moditect/layrry/config/UrlDownloader.java

```diff
@@ -0,0 +1,176 @@
+                protected PasswordAuthentication getPasswordAuthentication() {
+                    if (getRequestorType() == Authenticator.RequestorType.PROXY) {
+                        String protocol = getRequestingProtocol().toLowerCase();
+                        String host = properties.getProperty(protocol + ".proxyHost", "");
+                        String port = properties.getProperty(protocol + ".proxyPort", "80");
+                        String username = properties.getProperty(protocol + ".proxyUser", "");
+                        String password = properties.getProperty(protocol + ".proxyPassword", "");
+                        if (getRequestingHost().equalsIgnoreCase(host) &&
+                            Integer.parseInt(port) == getRequestingPort()) {
+                            return new PasswordAuthentication(username, password.toCharArray());
+                        }
+                    }
+                    return null;
+                }
+            });
+        }
+
+        if (getBoolean("socks.proxy", properties)) {
+            System.setProperty("socksProxyHost", properties.getProperty("socks.proxyHost"));
+            System.setProperty("socksProxyPort", properties.getProperty("socks.proxyPort"));
+            if (getBoolean("socks.proxy.auth", properties)) {
+                String username = properties.getProperty("socks.proxyUser", "");
+                String password = properties.getProperty("socks.proxyPassword", "");
+                System.setProperty("java.net.socks.username", username);
+                System.setProperty("java.net.socks.password", password);
+                Authenticator.setDefault(new Authenticator() {
+                    @Override
+                    protected PasswordAuthentication getPasswordAuthentication() {
+                        return new PasswordAuthentication(username, password.toCharArray());
+                    }
+                });
+            }
+        }
+    }
+
+    static boolean getBoolean(String name, Properties properties) {
+        try {
+            return Boolean.parseBoolean(properties.getProperty(name));
+        } catch (IllegalArgumentException | NullPointerException e) {
+            // ignored
+        }
+        return false;
+    }
+
+    static int getInteger(String name, Properties properties, int defaultValue) {
+        try {
+            return Integer.parseInt(properties.getProperty(name));
+        } catch (NumberFormatException | NullPointerException e) {
+            // ignored
+        }
+        return defaultValue;
+    }
+
+    /**
+     * Transfers the contents of the {@link InputStream} into the {@link OutputStream}, optionally closing the stream.
+     *
+     * @param istream the input stream
+     * @param ostream the output stream
+     * @param close   whether or not to close the output stream
+     */
+    static void transfer(InputStream istream, OutputStream ostream, final boolean close) throws IOException {
```

Comment:
Anything speaking against using [InputStream#transferTo()](https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/io/InputStream.html#transferTo(java.io.OutputStream))?

Reply:
I completely forgot about that method existed. Will certainly use a JSL method over a custom one.